projects/sandbox/datagen/datagen/scripts/timeslides.py

```diff
@@ -305,6 +310,19 @@ def main(
                 future = submit_write(pool, injection_ts, t, **injected_data)
                 futures.append(future)
 
+                # infer luminosity distance based on redshift and cosmology
+                parameters["luminosity_distance"] = cosmo.luminosity_distance(
```

Comment:
As long as we're always inheriting from a cosmologically-aware `bilby` prior, we can skip this step - `bilby` will calculate `luminosity_distance` from `redshift` if the former doesn't exist. On the other hand, this enforces that our desired cosmology is used. `bilby` will use the cosmology specified in the `redshift` prior, but if that's ever not set, it will default to Planck 15 (which is currently what you've set, but in case that ever changes, this would add backup)

Reply:
Ah okay. So will the `luminosity_distance` field automatically be populated?

Reply:
Yup, during sampling

Reply:
Correcting myself here as well as on Slack for completeness: the behavior I described doesn't happen. From the documentation, I believe that it's supposed to, but it does not. This conversion should stay.